daemon/logger/plugin_unix.go

```diff
@@ -12,7 +12,10 @@ import (
 )
 
 func openPluginStream(a *pluginAdapter) (io.WriteCloser, error) {
-	f, err := fifo.OpenFifo(context.Background(), a.fifoPath, unix.O_WRONLY|unix.O_CREAT|unix.O_NONBLOCK, 0700)
+	// open with read/write so avoid broken pipes in case of a plugin failure.
+	// It is up to the plugin to keep track of pipes that it should re-attach to, however.
+	// If the plugin doesn't open for reads, then the container will block once the pipe is full.
+	f, err := fifo.OpenFifo(context.Background(), a.fifoPath, unix.O_RDWR|unix.O_CREAT|unix.O_NONBLOCK, 0700)
```

Comment:
Not strictly related to this PR, but containerd repos all use OpenFifo with permMode set to `0` and not `0700`. Any ideas why?

Reply:
Not sure, some of it is `0` and some is `0700`

```
vendor/github.com/containerd/containerd/cio/io_unix.go:114:      if f.Stdin, err = fifo.OpenFifo(ctx, fifos.Stdin, syscall.O_WRONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0700); err != nil {
vendor/github.com/containerd/containerd/cio/io_unix.go:124:      if f.Stdout, err = fifo.OpenFifo(ctx, fifos.Stdout, syscall.O_RDONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0700); err != nil {
vendor/github.com/containerd/containerd/cio/io_unix.go:134:      if f.Stderr, err = fifo.OpenFifo(ctx, fifos.Stderr, syscall.O_RDONLY|syscall.O_CREAT|syscall.O_NONBLOCK, 0700); err != nil {
vendor/github.com/containerd/containerd/linux/proc/exec.go:166:      sc, err := fifo.OpenFifo(ctx, e.stdio.Stdin, syscall.O_WRONLY|syscall.O_NONBLOCK, 0)
vendor/github.com/containerd/containerd/linux/proc/init.go:203:      sc, err := fifo.OpenFifo(context, r.Stdin, syscall.O_WRONLY|syscall.O_NONBLOCK, 0)
vendor/github.com/containerd/containerd/linux/proc/init_state.go:202:        sc, err := fifo.OpenFifo(ctx, sio.Stdin, syscall.O_WRONLY|syscall.O_NONBLOCK, 0)
vendor/github.com/containerd/containerd/linux/proc/io.go:69:     fw, err := fifo.OpenFifo(ctx, name, syscall.O_WRONLY, 0)
vendor/github.com/containerd/containerd/linux/proc/io.go:73:     fr, err := fifo.OpenFifo(ctx, name, syscall.O_RDONLY, 0)
vendor/github.com/containerd/containerd/linux/proc/io.go:83: f, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
vendor/github.com/containerd/containerd/linux/shim/service_linux.go:45:      in, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
vendor/github.com/containerd/containerd/linux/shim/service_linux.go:58:  outw, err := fifo.OpenFifo(ctx, stdout, syscall.O_WRONLY, 0)
vendor/github.com/containerd/containerd/linux/shim/service_linux.go:62:  outr, err := fifo.OpenFifo(ctx, stdout, syscall.O_RDONLY, 0)
vendor/github.com/containerd/containerd/linux/shim/service_unix.go:36:       in, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
vendor/github.com/containerd/containerd/linux/shim/service_unix.go:49:   outw, err := fifo.OpenFifo(ctx, stdout, syscall.O_WRONLY, 0)
vendor/github.com/containerd/containerd/linux/shim/service_unix.go:53:   outr, err := fifo.OpenFifo(ctx, stdout, syscall.O_RDONLY, 0)
```